lib/pure/random.nim

```diff
@@ -276,8 +253,10 @@ proc rand*(r: var Rand; max: range[0.0 .. high(float)]): float {.benign.} =
     let u = (0x3FFu64 shl 52u64) or (x shr 12u64)
     result = (cast[float](u) - 1.0) * max
 
-proc rand*(max: float): float {.benign.} =
-  ## Returns a random floating point number in the range `0.0..max`.
+proc rand*[T: Ordinal or SomeFloat](max: T): T {.benign.} =
+  ## Returns a random floating point number in the range `T(0) .. max`.
+  ## 
+  ## Allowed types for `T` are integers, floats, and enums without holes.
```

Comment:
this comment is redundant (and only approximately correct) and should be removed, it's already specified what T is in the generated docs:

![image](https://user-images.githubusercontent.com/2194784/101419744-83fbab00-38a5-11eb-9d76-99ab7bd23dba.png)

furthermore, `Ordinal` is clickable (thanks to docgen) so users who see `Ordinal` for the 1st time can just click to see the definition, which is explained in 1 place, where it belongs (https://nim-lang.github.io/Nim/system.html#Ordinal) => "1 definition rule" for docs

note that the definition for https://nim-lang.github.io/Nim/system.html#Ordinal should be fixed via `s/enums/enums without holes/` but that's a separate issue